mne/beamformer/_lcmv.py

```diff
@@ -39,28 +39,36 @@ def _reg_pinv(x, reg):
     return linalg.pinv(x), d
 
 
-def _setup_picks(picks, info, forward, noise_cov=None):
-    if picks is None:
-        picks = pick_types(info, meg=True, eeg=True, ref_meg=False,
-                           exclude='bads')
-
-    ok_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+def _setup_picks(info, forward, data_cov=None, noise_cov=None):
+    # get a list of all channel names:
+    fwd_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+
+    # handle channels from forward model and info:
+    ch_names = [c['ch_name'] for c in info['chs']
+                if c['ch_name'] not in info['bads'] and
+                c['ch_name'] in fwd_ch_names]
+
+    # handle channels from data cov if data cov is not None
+    # Note: data cov is supposed to be None in tf_lcmv
+    if data_cov is not None:
+        ch_names = [c for c in ch_names
+                    if c not in data_cov['bads'] and
```

Comment:
do you think we should force the bads in the noise cov to be exactly the same as the bads in the data? It does not matter if the noise cov is not regularized but otherwise it's likely to be problematic. 

thoughts? ping @dengemann @Eric89GXL